app/javascript/lib/validation/validate-file-content.js

```diff
@@ -220,6 +236,70 @@ function validateValuesAreNumeric(line, isLastLine, lineNum, dataObj) {
     `Please ensure all values in ${rowText}: ${notedBadRows}, are numbers.`
     issues.push(['error', 'format:invalid-type:not-numeric', msg])
   }
+
+  return issues
+}
+
+/**
+ * Verify sparse matrix column numbers match
+ */
+function validateSparseColumnNumber(line, isLastLine, lineNum, dataObj) {
```

Comment:
there might be a case for refactoring this into a more general `validateColumnNumbers(numExpectedCols, line, isLastLine, lineNum, dataObj)`. but that can be done later.